### PR TITLE
(#3505) - fix sqlite plugin support

### PR DESCRIPTION
**Pull request by nolanlawson**
My goal: to get PouchDB to use the SQLite Plugin without any manual intervention from the user.

I succeeded! There were just a few things to change:

* `window.sqlitePlugin` is not declared until after Cordova's `deviceready` event has fired, so we cannot use it to detect `valid()`. (I.e. because of Windows Phone 8, which has sqlitePlugin support but not `openDatabase`.) But luckily the SQLitePlugin leaks a global `SQLitePlugin` variable, so we can use that to detect SQLitePlugin support. Flakey, but it fixes Windows Phone 8.
* the `openDB` function has to be re-generated every time a new database is created, because the global variables might have changed post-`deviceready`, in which case we want to use the SQLite plugin instead of `openDatabase`.

So basically what this requires users to do is to not call `new PouchDB()` until after the `deviceready` event (which they are probably accustomed to doing. I verified in [my sample app](https://github.com/nolanlawson/pouchdb-cordova-hello-world-with-sqlite-plugin) that it was actually working and actually using the SQLite Plugin.

### CODE REVIEW

File: lib/adapters/websql/websql.js [hunk 1]
```diff
@@ -60,28 +61,6 @@ function fetchAttachmentsIfNecessary(doc, opts, api, txn, cb) {
   });
 }
 
-var cachedDatabases = {};
-
-var sqlitePluginOpenDBFunction =
-  ((typeof navigator !== 'undefined' &&
-    navigator.sqlitePlugin &&
-    navigator.sqlitePlugin.openDatabase) &&
-    navigator.sqlitePlugin.openDatabase.bind(navigator.sqlitePlugin)) ||
```
Comment (nolanlawson) [inline lib/adapters/websql/websql.js]: This `navigator.sqlitePlugin` stuff seems to date from an older version of the SQLite Plugin. I'm happy to just require that users use the latest vesion of the SQLite Plugin.

**Comment by daleharvey** [comment]
Looks great, cheers

https://github.com/pouchdb/pouchdb/commit/7a7ee7d1dcad2ddfb79baaf4bc00b7d3551df8c4
